### PR TITLE
{vis}[foss/2018b] Graphviz v2.40.1

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    ('Pango', '1.42.4'),
+    ('Perl', '5.28.0'),
+    ('Qt5', '5.10.1'),
+    ('Tcl', '8.6.8'),
+    ('zlib', '1.2.11'),
+]
+
+builddependencies = [
+    ('Autotools', '20180311'),
+    ('SWIG', '3.0.12', '-Python-3.6.6'),
```
Comment (boegel) [inline easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: @fizwit You can't mix Python 2.x and Python 3.x in a single dependency tree, so this should be changed to use the Python 2.7.15 version of SWIG:

```python
('SWIG', '3.0.12', '-Python-2.7.15'),
```

(that's also why the tests currently fail due to a conflict problem)

File: easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ learning, and in visual interfaces for other technical domains."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+source_urls = ['https://graphviz.gitlab.io/pub/graphviz/stable/SOURCES/']
+sources = ['graphviz.tar.gz']
+checksums = ['ca5218fade0204d59947126c38439f432853543b0818d9d728c589dfe7f3a421']
+
+python = "Python"
+pythonversion = '2.7.15'
```
Comment (boegel) [inline easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: @fizwit Neither of these are used, so please remove these two lines?

File: easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+description = """Graphviz is open source graph visualization software. Graph
+ visualization is a way of representing structural information as diagrams of
+ abstract graphs and networks. It has important applications in networking,
+ bioinformatics,  software engineering, database and web design, machine
+ learning, and in visual interfaces for other technical domains."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+source_urls = ['https://graphviz.gitlab.io/pub/graphviz/stable/SOURCES/']
+sources = ['graphviz.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: Please use the alternate source URL we use in the existing easyconfig for `Graphviz` 2.40.1, see https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-intel-2018a.eb#L15

Comment (fizwit) [reply easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: source_urls changed
source_urls = ['https://fossies.org/linux/misc/']

File: easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    ('Autotools', '20180311'),
+    ('SWIG', '3.0.12', '-Python-2.7.15'),
+    ('pkg-config', '0.29.2'),
+]
+
+preconfigopts = './autogen.sh && '
+
+configopts = '--enable-guile=no --enable-lua=no --enable-ocaml=no '
+configopts += '--enable-r=no --enable-ruby=no '
+configopts += '--enable-ruby=no --enable-php=no '
```
Comment (vanzod) [inline easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: @fizwit `--enable-ruby=no` is duplicated

File: easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+description = """Graphviz is open source graph visualization software. Graph
+ visualization is a way of representing structural information as diagrams of
+ abstract graphs and networks. It has important applications in networking,
+ bioinformatics,  software engineering, database and web design, machine
+ learning, and in visual interfaces for other technical domains."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+source_urls = ['https://fossies.org/linux/misc/']
+sources = ['graphviz.tar.gz']
```
Comment (vanzod) [inline easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: For consistency, let's use `sources = [SOURCELOWER_TAR_GZ]` here

File: easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb (hunk 4 above)
Comment (fizwit) [reply easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: ok, removing unnecessary ruby

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 3 out of 3 (1 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/3814ba178a8384937cd414ad6569ec20 for a full test report.

File: easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+prebuildopts = 'qmake -o cmd/gvedit/qMakefile cmd/gvedit/gvedit.pro && '
+
+sanity_check_paths = {
+    'files': ['bin/cluster', 'bin/dot', 'bin/gvmap',
+              'lib/libcdt.%s' % SHLIB_EXT,
+              'lib/libgvc.%s' % SHLIB_EXT,
+              'lib/libxdot.%s' % SHLIB_EXT],
+    'dirs': ['include']
+}
+
```
Comment (vanzod) [inline easybuild/easyconfigs/g/Graphviz/Graphviz-2.40.1-foss-2018b.eb]: @fizwit Please add the following `sanity_check_commands` for consistency with previous easyconfig files:
```
sanity_check_commands = [
    ("test ! -d $EBROOTTCL/lib/*/graphviz", ''),
    ("test ! -d $EBROOTTCL/lib64/*/graphviz", ''),
]
```

**Pull request closed without merging**